src/CTA.Rules.Actions/ClassActions.cs

```diff
@@ -152,5 +152,23 @@ public class ClassActions
             };
             return RenameClass;
         }
+        public Func<SyntaxGenerator, ClassDeclarationSyntax, ClassDeclarationSyntax> GetReplaceMethodModifiersAction(string methodName, string modifiers)
+        {
+            Func<SyntaxGenerator, ClassDeclarationSyntax, ClassDeclarationSyntax> ReplaceMethodModifiers = (SyntaxGenerator syntaxGenerator, ClassDeclarationSyntax node) =>
+            {
+                var allMembers = node.Members.ToList();
+                var allMethods = allMembers.OfType<MethodDeclarationSyntax>();
+                var replaceMethod = allMethods.Where(m => m.Identifier.ToString() == methodName).FirstOrDefault();
+                allMembers.Remove(replaceMethod);
+                SyntaxTokenList tokenList = new SyntaxTokenList(SyntaxFactory.ParseTokens(modifiers));
+                var newMethod = replaceMethod.WithModifiers(tokenList);
+                allMembers.Add(replaceMethod);
+
+                node = node.WithMembers(node.Members.Replace(replaceMethod, newMethod)).NormalizeWhitespace();
```

Comment:
Why are we modifying allMembers if we're just replacing it. We can either use allMembers inside the WithMembers, or not modify it.

Reply:
Copied from the remove action, I went ahead and fixed that one too :-) thanks.